exporting/exporting_engine.c

```diff
@@ -4,6 +4,55 @@
 
 static struct engine *engine = NULL;
 
+/**
+ * Exporting Clean Engine
+ *
+ * Clean all variables allocated inside engine structure
+ *
+ * @param en a pointer to the strcuture that will be cleaned.
+ */
+static void exporting_clean_engine()
+{
+    if (!engine)
+        return;
+
+    int disable_awk_sdk = 0;
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+    int disable_remote_write = 0;
+#endif
+    for (struct instance *instance = engine->instance_root; instance;) {
+        struct instance *current_instance = instance;
+        instance = instance->next;
+
+        if (current_instance->config.type == EXPORTING_CONNECTOR_TYPE_PROMETHEUS_REMOTE_WRITE) {
+#if ENABLE_PROMETHEUS_REMOTE_WRITE
+            clean_prometheus_remote_write_instance(instance);
+            if (!disable_remote_write) {
+                protocol_buffers_shutdown();
+                disable_remote_write = 1;
+            }
+#endif
+        } else if (current_instance->config.type == EXPORTING_CONNECTOR_TYPE_KINESIS
+                 && current_instance->engine->aws_sdk_initialized && !disable_awk_sdk) {
+#if HAVE_KINESIS
+            aws_sdk_shutdown();
+#endif
+            disable_awk_sdk = 1;
+        }
+
+        clean_instance(current_instance);
+    }
+
+    //Cleanup web api
+    prometheus_clean_server_root();
+
+    if (engine->config.prefix)
+        freez((void *)engine->config.prefix);
+    if (engine->config.hostname)
```

Comment:
Excessive check.

Reply:
This is a habit, I removed the checks before all `freez`.